modules/ROOT/pages/deployment/storage/general-considerations.adoc

```diff
@@ -0,0 +1,173 @@
+
+{empty} +
+
+[role=center,width=70%,cols="40%,35%,80%",options="header"]
+|===
+| Envorinment Variable
+| Target
+| Space requirement
+
+| `OCIS_CONFIG_DIR`
+| NFS_1
+| Very low
+
+| `OCIS_BASE_DATA_PATH`
+| NFS_2
+| High
+
+| Seach and Thumbnails
+| NFS_3
+| Medium, to be monitored
+|===
+
+== S3
+
+S3 to store blobs is typically used by large enterprises and hosters, though it can fit for medium enterprises too. Data distribution and separation is a bit different compared to a pure POSIX backend. For details on S3 see the xref:deployment/storage/s3.adoc[S3] documentation. Also see the xref:deployment/general/general-info.adoc#using-s3-for-blobs[Using S3 for Blobs] description:
+
+* POSIX storage, usually NFS.
+** Metadata
+** Data for search and/or thumbnails
+** Other data
+
+* S3 for blobs
+
+As you can see in the table below, the layout distributes data over different storages and mounts based on their use case. This is by nature to enable scaling the system according the needs of large enterprises.
+
+{empty} +
+
+[role=center,width=90%,cols="40%,45%,80%",options="header"]
+|===
+| Envorinment Variable
+| Target
+| Space requirement
+
+| `OCIS_CONFIG_DIR`
+| NFS_1
+| Very low
+
+| `OCIS_BASE_DATA_PATH`
+| NFS_2
+| Medium, to be monitored +
+Alternatively low if NFS_4
+
+| Seach and Thumbnails
+| NFS_3
+| Medium, to be monitored
+
+| `STORAGE_USERS_S3NG_ROOT`
+| `OCIS_BASE_DATA_PATH` +
+Alternatively NFS_4
+| Medium, to be monitored
+
+| S3 specific settings
+| S3
+| High
+|===
+
+== Resource Optimisation
+
+Depending on the storage connected and the servers capabilities, Infinite Scale can be optimized using the servers resources. The relevant environment variable to configure this is:
+
+`STORAGE_USERS_OCIS_MAX_CONCURRENCY`
+
+The value to consider is based on how much CPU's and memory the server has the instance of the xref:{s-path}/storage-users.adoc[storage-users] service is running on, which kind of storage, POSIX or S3 is used for blobs and what you want to achieve.
+
+=== Background
+
+In a nutshell, the value for `STORAGE_USERS_OCIS_MAX_CONCURRENCY` defines how many workers are assigned to storage related tasks. Any worker not only serves its job, but also consumes CPU and memory resources which needs to balance out. On the other hand side, when it comes to the connected storage, workers serving S3 will be more in response waiting time compared to POSIX connections. As workers which are in waiting state do consume less resources, the value can be considered to allow overcommitting CPU resources.
+
+=== Value Considerations
+
+The following considerations are based on using POSIX storage only
+
+* Performance without worrying about memory +
+`runtime.NumCPU() * 2`
+* Performance +
+`runtime.NumCPU()`
+* Limited memory available +
+ A value of 4 or lower, assuming 4 is still lower than the number of CPU available
```

Comment:
These are assumptions from CS-courses I do not recommend using them in an official documentation. (Though I do not have any alternatives atm).

Reply:
As discussed, we let the terms for the time being.
I additionally updated the text as discussed.